app/javascript/hooks/mutations/admin/roles/useDeleteRole.jsx

```diff
@@ -12,7 +12,13 @@ export default function useDeleteRole({ role, onSettled }) {
   return useMutation(
     () => axios.delete(`/admin/roles/${role.id}.json`),
     {
-      onError: () => { toast.error(t('toast.error.problem_completing_action')); },
+      onError: (error) => {
+        if (error.response?.status === 500) {
+          toast.error(t('toast.error.roles.role_assigned'));
```

Comment:
I'd vote against this,
An internal server error `500` isn't limited to one singular specific case,
If we wanna make sense of the case where a failed delete attepmt is because a role is still in use (which I believe what we should do to avoid giving false negatives which gives a terrible UX).
The API need to be specific by returning something that is in that context unique to this scenario wither through a unique response status code or if it's not feasible or meaninful it can provide an identifier in the response data,

Reply:
Would something like 403 Forbidden be more acceptable?

Reply:
Maybe, but I think that it's used in the same context to indicate insufficient permissions?

Reply:
405 Method Not Allowed would be perfect for this case.

Reply:
Umm, it seems to be used natively by the web servers and clients.
Was wondering about the implication of the direct usage of it without specifying the `Allow` header,
But after some searching it does not seem to be an issue,
So yeah we can use that,